File: XML/SAX/sax.py
#!/usr/bin/python
import xml.sax
 
class ABContentHandler(xml.sax.ContentHandler):
  def __init__(self):
    xml.sax.ContentHandler.__init__(self)
 
  def startElement(self, name, attrs):
    print("<" + name + ">"),
    for i in attrs.getNames():
      print(attrs.getValue(i) + ", "),
    
  def endElement(self, name):   
    print("</" + name + ">")
 
  def characters(self, content):
    print(content),
 
def main(sourceFileName):
  source = open(sourceFileName)
  xml.sax.parse(source, ABContentHandler())
 
if __name__ == "__main__":
  main("recipes.xml")

File: XML/SAX/SAX.java

import org.xml.sax.*;
import org.xml.sax.helpers.*;
import org.apache.xerces.parsers.SAXParser;

public class SAX extends DefaultHandler {
    @Override
    public void startElement(String namespaceURI, String localName, String qName, Attributes atts) {
        System.out.print("<" + qName + "> ");
        for (int i = 0; i < atts.getLength(); i++) {
            System.out.print(atts.getValue(i) + ", ");
        }
    }

    @Override
    public void endElement(String uri, String localName, String qName)
            throws SAXException {
        System.out.println("</" + qName + "> ");
    }

    @Override
    public void characters(char ch[], int start, int length) throws SAXException {
        if (length > 5) {
            System.out.print(new String(ch, start, length));
        }
    }

    public static void main(String[] args) {
        SAX f = new SAX();
        SAXParser p = new SAXParser();
        p.setContentHandler(f);

        try {
            p.parse("recipes.xml");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
